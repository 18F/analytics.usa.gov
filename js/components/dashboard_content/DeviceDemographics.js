import React from "react";
import PropTypes from "prop-types";

import DevicesChart from "./DevicesChart";
import BrowsersChart from "./BrowsersChart";
import OperatingSystemsChart from "./OperatingSystemsChart";
import ConsolidatedBarChart from "../chart/ConsolidatedBarChart";

/**
 * Contains charts and other data visualizations for the user demographics
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function DeviceDemographics({ dataHrefBase }) {
  return (
    <div className="padding-top-4 grid-row">
      <div id="devices" className="desktop:grid-col-4 desktop:padding-right-2">
        <DevicesChart dataHrefBase={dataHrefBase} />
        <div className="chart__title desktop:padding-top-2">
          <a href="/definitions#dimension_screen_resolution">
            Top Screen Resolutions
          </a>
        </div>
        <ConsolidatedBarChart
          dataUrl={`${dataHrefBase}/screen-resolution-30-days.json`}
          chartDataKey={"screen_resolution"}
          maxItems={10}
        />
      </div>

      <div id="browsers" className="desktop:grid-col-4 desktop:padding-x-2">
        <BrowsersChart dataHrefBase={dataHrefBase} />
      </div>

      <div
        id="operating_systems"
        className="desktop:grid-col-4 desktop:padding-left-2"
      >
        <OperatingSystemsChart dataHrefBase={dataHrefBase} />
      </div>
    </div>
  );
}

DeviceDemographics.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default DeviceDemographics;
