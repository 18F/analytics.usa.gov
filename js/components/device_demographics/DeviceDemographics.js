import React from "react";
import PropTypes from "prop-types";

import DevicesChart from "./DevicesChart";
import BrowsersChart from "./BrowsersChart";
import OperatingSystemsChart from "./OperatingSystemsChart";

function DeviceDemographics({ dataURL, dataPrefix }) {
  const dataHrefBase = `${dataURL}/${dataPrefix}`;

  return (
    <>
      <section id="devices" className="desktop:grid-col-4 bar-chart-component">
        <h4>Devices</h4>
        <DevicesChart dataHrefBase={dataHrefBase} />
      </section>

      <section id="browsers" className="desktop:grid-col-4 bar-chart-component">
        <h4>Web Browsers</h4>
        <BrowsersChart dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="operating_systems"
        className="desktop:grid-col-4 bar-chart-component"
      >
        <h4>Operating Systems</h4>
        <OperatingSystemsChart dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

DeviceDemographics.propTypes = {
  dataURL: PropTypes.string.isRequired,
  dataPrefix: PropTypes.string.isRequired,
};

export default DeviceDemographics;
