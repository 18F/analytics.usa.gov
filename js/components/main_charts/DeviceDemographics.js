import React from "react";
import PropTypes from "prop-types";

import DevicesChart from "./DevicesChart";
import BrowsersChart from "./BrowsersChart";
import OperatingSystemsChart from "./OperatingSystemsChart";

/**
 * Contains charts and other data visualizations for the user demographics
 * section of the site. This component is mainly laying out the structure for
 * the section and passes props necessary for getting data and displaying
 * visualizations to child components.
 *
 * @param {object} props the properties for the component
 * @param {string} props.dataHrefBase the URL of the base location of the data
 * to be downloaded including the agency path. In production this is proxied and
 * redirected to the S3 bucket URL.
 * @returns {import('react').ReactElement} The rendered element
 */
function DeviceDemographics({ dataHrefBase }) {
  return (
    <>
      <section id="devices" className="desktop:grid-col-4 bar-chart-component">
        <h4>Devices</h4>
        <DevicesChart dataHrefBase={dataHrefBase} />
      </section>

      <section id="browsers" className="desktop:grid-col-4 bar-chart-component">
        <h4>Web Browsers</h4>
        <BrowsersChart dataHrefBase={dataHrefBase} />
      </section>

      <section
        id="operating_systems"
        className="desktop:grid-col-4 bar-chart-component"
      >
        <h4>Operating Systems</h4>
        <OperatingSystemsChart dataHrefBase={dataHrefBase} />
      </section>
    </>
  );
}

DeviceDemographics.propTypes = {
  dataHrefBase: PropTypes.string.isRequired,
};

export default DeviceDemographics;
