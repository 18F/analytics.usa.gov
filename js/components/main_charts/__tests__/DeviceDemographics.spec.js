import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import DeviceDemographics from "../DeviceDemographics";

describe("DeviceDemographics", () => {
  let component;

  beforeEach(async () => {
    component = await render(
      <DeviceDemographics dataHrefBase="http://www.example.com/data/" />,
    );
    await waitFor(() => screen.getByText("Web Browsers"));
  });

  it("renders", () => {
    expect(component.asFragment()).toMatchSnapshot();
  });
});
