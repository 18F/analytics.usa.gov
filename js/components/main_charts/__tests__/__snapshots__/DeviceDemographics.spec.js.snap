// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceDemographics renders 1`] = `
<DocumentFragment>
  <section
    class="desktop:grid-col-4 bar-chart-component"
    id="devices"
  >
    <h4>
      Devices
    </h4>
    <figure
      id="chart_device_types"
    >
      <div
        class="data bar-chart"
      />
    </figure>
  </section>
  <section
    class="desktop:grid-col-4 bar-chart-component"
    id="browsers"
  >
    <h4>
      Web Browsers
    </h4>
    <figure
      id="chart_browsers"
    >
      <div
        class="data bar-chart"
      />
    </figure>
  </section>
  <section
    class="desktop:grid-col-4 bar-chart-component"
    id="operating_systems"
  >
    <h4>
      Operating Systems
    </h4>
    <div>
      <figure
        id="chart_os"
      >
        <div
          class="data bar-chart"
        />
      </figure>
      <figure
        class="hide"
        data-scale-to-parent="true"
        id="chart_windows"
      >
        <h4>
          Windows
        </h4>
        <div
          class="data bar-chart"
        />
      </figure>
    </div>
  </section>
</DocumentFragment>
`;
